### PR TITLE
ocamlPackages.netchannel: 2.0.0 → 2.1.1; ocamlPackages.mirage-xen: 6.0.0 → 7.2.0

### PR DESCRIPTION
**Pull request by vbgl**
###### Description of changes

Compatibility with recent `cstruct`.

cc maintainer @sternenseemann 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/netchannel/default.nix [hunk 1]
```diff
@@ -17,15 +17,13 @@
 
 buildDunePackage rec {
```
Comment (sternenseemann) [inline pkgs/development/ocaml-modules/netchannel/default.nix:18]: Removed dependencies:

* rresult

New dependencies (probably propagated already?):

* macaddr
* lwt-dllist
* result

File: pkgs/development/ocaml-modules/mirage-xen/default.nix [hunk 2]
```diff
@@ -19,28 +19,14 @@
...
-    substituteInPlace lib/bindings/mirage-xen.pc \
-      --replace "@out@" "$out" \
-      --replace "@OCAMLFIND_DESTDIR@" "$OCAMLFIND_DESTDIR"
-  '';
-
-  minimumOCamlVersion = "4.08";
+  minimalOCamlVersion = "4.08";
 
   nativeBuildInputs = [
     pkg-config
```
Comment (sternenseemann) [inline pkgs/development/ocaml-modules/mirage-xen/default.nix:32]: pkg-config should be unnecessary now.

File: pkgs/development/ocaml-modules/mirage-xen/default.nix [hunk 3]
```diff
@@ -62,13 +48,6 @@ buildDunePackage rec {
     (ocaml-freestanding.override { target = "xen"; })
```
Comment (sternenseemann) [inline pkgs/development/ocaml-modules/mirage-xen/default.nix:48]: ocaml-freestanding should be unnecessary now.

**Comment by vbgl** [comment]
Thanks for the review! I’ve done the suggested changes.

**Pull request merged**